### PR TITLE
Retrofit shippable scripts to work with AZP

### PR DESCRIPTION
**Pull request by eqrx**
##### SUMMARY
Scripts under `hacking/shippable` are still only compatible with shippable, even though we switched to AZP. This PR makes them the scripts work with AZP.

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

File: hacking/shippable/incidental.py [hunk 1]
```diff
@@ -132,14 +137,9 @@ def incidental_report(args):
...
-    if coverage_data.status_code != 30:
-        check_failed(args, 'results from Shippable indicate tests did not pass (status code: %d)\n'
-                           're-run until passing, then download the latest results and re-run the report using those results' % coverage_data.status_code)
-
-    if coverage_data.missing_jobs or coverage_data.extra_jobs:
-        check_failed(args, 'unexpected results from Shippable -- missing jobs: %s, extra jobs: %s\n'
-                           'make sure the tests were successful and the all results were downloaded\n' % (
-                               sorted(coverage_data.missing_jobs), sorted(coverage_data.extra_jobs)))
+    if coverage_data.result != "succeeded":
+        check_failed(args, 'results from Shippable indicate tests did not pass (result: %s)\n'
```
Comment (mattclay) [inline hacking/shippable/incidental.py]:
```suggestion
        check_failed(args, 'results indicate tests did not pass (result: %s)\n'
```

File: hacking/shippable/download.py [hunk 2]
```diff
@@ -35,22 +38,33 @@
...
 
 def main():
     """Main program body."""
+
     args = parse_args()
     download_run(args)
 
 
+def run_id_arg(arg):
+    m = re.fullmatch(r"(?:https:\/\/dev\.azure\.com\/ansible\/ansible\/_build\/results\?buildId=)?(\d{4})", arg)
```
Comment (mattclay) [inline hacking/shippable/download.py]:
```suggestion
    m = re.fullmatch(r"(?:https:\/\/dev\.azure\.com\/ansible\/ansible\/_build\/results\?buildId=)?(\d+)", arg)
```

File: hacking/shippable/rebalance.py [hunk 3]
```diff
@@ -22,11 +22,13 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+import datetime
```
Comment (mattclay) [inline hacking/shippable/rebalance.py]: Omit the changes to this script. The JSON files it used to rebalance targets are not currently saved as Azure Pipelines artifacts, so the necessary data is currently unavailable. Once that's resolved this script can be updated in a separate PR.

**Review by mattclay — CHANGES_REQUESTED** [review]

File: hacking/shippable/download.py [hunk 4]
```diff
@@ -62,28 +76,12 @@ def parse_args():
                         action='store_true',
                         help='show what would be downloaded without downloading')
 
-    parser.add_argument('--key',
-                        dest='api_key',
-                        default=api_key,
-                        required=api_key is None,
-                        help='api key for accessing Shippable')
-
-    parser.add_argument('--console-logs',
```
Comment (mattclay) [inline hacking/shippable/download.py:71]: Can you restore this functionality to the script? Although it's not used for incidental coverage, it does get used for other tasks.

Comment (eqrx) [reply hacking/shippable/download.py:71]: Sure, on it

File: hacking/shippable/download.py [hunk 5]
```diff
@@ -92,35 +94,25 @@ def parse_args():
                         action='store_true',
                         help='download everything')
 
-    parser.add_argument('--job-number',
-                        metavar='N',
-                        action='append',
-                        type=int,
-                        help='limit downloads to the given job number')
+    parser.add_argument('--match-jobs',
+                        default=re.compile('.*CentOS.*'),
```
Comment (mattclay) [inline hacking/shippable/download.py]: The default should match all jobs.

File: hacking/shippable/download.py [hunk 6]
```diff
@@ -205,181 +145,34 @@ def download_run(args):
...
-        with open(path, 'r') as key_fd:
-            return key_fd.read().strip()
-    except IOError:
-        return None
+    if args.coverage_results:
+        artifact_list_url = 'https://dev.azure.com/ansible/ansible/_apis/build/builds/%s/artifacts?api-version=6.0' % args.run
+        artifact_list_response = requests.get(artifact_list_url)
+        artifact_list_response.raise_for_status()
+        for artifact in artifact_list_response.json()['value']:
+            if not args.match_jobs.match(artifact['name']):
```
Comment (mattclay) [inline hacking/shippable/download.py]: This is performing a job match against the artifact name. In order to filter artifacts based on jobs you need to use the timeline results and match the artifact `source` against the timeline `id` for a given job.

File: hacking/shippable/download.py [hunk 7]
```diff
@@ -205,181 +145,34 @@ def download_run(args):
...
-        return key
-
-    path = os.path.join(os.environ['HOME'], '.shippable.key')
-
-    try:
-        with open(path, 'r') as key_fd:
-            return key_fd.read().strip()
-    except IOError:
-        return None
+    if args.coverage_results:
```
Comment (mattclay) [inline hacking/shippable/download.py]: I'd rename this and the associated CLI option to `artifacts`, since not all artifacts are coverage artifacts.

**Review by mattclay — CHANGES_REQUESTED** [review]

**Review by mattclay — APPROVED** [review]
